src/frontend/js/components/SearchFilterGroupModal/index.tsx

```diff
@@ -54,118 +63,191 @@ const messages = defineMessages({
-      setValues([]);
       setQuery('');
       setError(null);
     }
   }, [modalIsOpen]);
 
-  useAsyncEffect(async () => {
-    // We can't start using full-text search until our text query is at least 3 characters long.
-    if (!modalIsOpen || (query.length > 0 && query.length < 3)) {
-      return;
-    }
+  // We need the current course search params to get the facet counts
+  const { courseSearchParams, dispatchCourseSearchParamsUpdate } = useCourseSearchParams();
 
-    const searchResponse = await fetchList(filter.name, {
-      limit: '20',
-      offset: '0',
-      query,
+  const fetchResults = async (args: any) => {
+    const { pageParam, queryKey } = args;
+    const [filterName, fetchCourseSearchParams, fetchQuery] = queryKey;
+    const searchResponse = await fetchList(filterName, {
+      ...(pageParam || API_LIST_DEFAULT_PARAMS),
+      ...(fetchQuery ? { query: fetchQuery } : {}),
     });
 
     if (searchResponse.status === RequestStatus.FAILURE) {
-      setValues([]);
       return setError(messages.error);
     }
 
     const facetResponse = await fetchList('courses', {
-      ...courseSearchParams,
-      [`${filter.name}_aggs`]: searchResponse.content.objects.map((resource) => resource.id),
+      ...fetchCourseSearchParams,
+      [`${filterName}_aggs`]: searchResponse.content.objects.map((resource) => resource.id),
       scope: 'filters',
+      facet_sorting: 'name',
     });
 
     if (facetResponse.status === RequestStatus.FAILURE) {
-      setValues([]);
       return setError(messages.error);
     }
 
-    const newValues = facetResponse.content.filters[filter.name].values;
+    return {
+      meta: searchResponse.content.meta,
+      objects: facetResponse.content.filters[filterName].values,
+    };
+  };
+
+  const { data, fetchNextPage, hasNextPage, isFetchingNextPage, status } = useInfiniteQuery(
+    [filter.name, courseSearchParams, query.length > 2 ? query : ''],
+    fetchResults,
+    {
+      enabled: modalIsOpen,
+      getNextPageParam: (lastPage) => {
+        if (!lastPage) {
+          return undefined;
+        }
+        const newOffset = lastPage.meta.offset + lastPage.meta.count;
+        if (newOffset >= lastPage.meta.total_count) {
+          return undefined;
+        }
+        return { limit: API_LIST_DEFAULT_PARAMS.limit, offset: newOffset };
+      },
+    },
+  );
 
-    setError(null);
-    setValues(newValues);
-  }, [modalIsOpen, query]);
+  const loadMoreButtonRef = useRef<HTMLButtonElement>(null);
+  useIntersectionObserver({
+    target: loadMoreButtonRef,
+    onIntersect: fetchNextPage,
+    enabled: !!hasNextPage,
+  });
+
+  return (
+    <Fragment>
+      <button className="search-filter-group-modal__close" onClick={() => setModalIsOpen(false)}>
+        <span className="offscreen">
+          <FormattedMessage {...messages.closeButton} />
+        </span>
+        <svg aria-hidden={true} role="img" className="icon search-filter-group-modal__close__icon">
```

Comment:
If you use `.modal_closeButton`, you have to use `.modal__closeButton__icon` class for the icon.
```suggestion
        <svg aria-hidden={true} role="img" className="modal__closeButton__icon">

```